packages/vm/src/evm/eei.ts

```diff
@@ -589,6 +589,10 @@ export default class EEI {
       return new BN(0)
     }
 
+    // EIP-2681 check
+    if (this._env.contract.nonce.eq(MAX_UINT64)) {
+      return new BN(0)
+    }
```

Comment:
Just for my own understanding:

I have to admit that I do not really understand what happens with this `BN(0)` at the end. So this will be put on the stack in the `CREATE` or `CREATE2` opcode functions. And then? Where is this evaluated and from what context and how does this come to a halt? 🤔

(I know this is a very general question which applies also to other contexts)

Reply:
Zero is put on stack in case if the contract creation fails (which happens here). In solidity, if you try to create a contract from your contract, then the compiler will also create a check to see if the contract was created correctly (i.e. the top stack item is nonzero). If that's the case, it reverts the entire frame. Note that in case of a succesfull `CREATE`/`CREATE2`, then the address of the just created contract is put on the stack. This frame comes to a halt because the [actual code to create the contract (the `message`)](https://github.com/ethereumjs/ethereumjs-monorepo/blob/55087ec44a0d1ca9ea90da152e83de8bf7dac78b/packages/vm/src/evm/eei.ts#L599) is never evaluated / executed.